compiler/rustc_const_eval/src/check_consts/check.rs

```diff
@@ -283,7 +283,10 @@ impl<'mir, 'tcx> Checker<'mir, 'tcx> {
                 safe_to_expose_on_stable,
                 is_function_call,
                 gate_already_checked,
-            } if gate_already_checked || self.tcx.features().enabled(gate) => {
+            } if gate_already_checked
+                || self.tcx.features().enabled(gate)
+                || span.allows_unstable(gate) =>
```

Comment:
@RalfJung I had to add this line in order to use the same feature gate for `ustable` and `rustc_const_unstable`

Reply:
No, we definitely do not want to check the span here, that could be used to bypass recursive const stability.

Reply:
Yes that's what I meant in https://github.com/rust-lang/rust/pull/138374#discussion_r2032731979. Now that you figured out why the separate feature gate is needed, it makes sense; we should just make sure to have a comment so our future selves still know this. :)